services/introspection/service.go

```diff
@@ -81,6 +90,58 @@ func (s *service) Plugins(ctx context.Context, req *api.PluginsRequest) (*api.Pl
 	}, nil
 }
 
+func (s *service) Server(ctx context.Context, _ *ptypes.Empty) (*api.ServerResponse, error) {
+	u, err := s.getUUID()
+	if err != nil {
+		return nil, errdefs.ToGRPC(err)
+	}
+	return &api.ServerResponse{
+		UUID:         u,
+		OS:           runtime.GOOS,
+		Architecture: runtime.GOARCH,
+	}, nil
+}
+
+func (s *service) getUUID() (string, error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	data, err := ioutil.ReadFile(s.uuidPath())
+	if err != nil {
+		if os.IsNotExist(err) {
+			return s.generateUUID()
+		}
+		return "", err
+	}
```

Comment:
error out if `data` contains extra whitespace or does not conform to the UUID spec?